### PR TITLE
added Dropdown submenu feature #205

### PR DESCRIPTION
**Pull request by plugn**
Hi! 
Here's sub-menus for Dropdown #205

I needed this new feature too and create it.  Now this templates are supported!
```
<template lang="html">

  <sui-dropdown text="File" item :is-open="true">
    <sui-dropdown-menu>
      <sui-dropdown-item>New</sui-dropdown-item>
      <sui-dropdown-item>Open... <span class="description">ctrl + o</span></sui-dropdown-item>

      <sui-dropdown-item>
        <sui-icon name="dropdown" /> Advanced

        <sui-dropdown-menu class="">
          <sui-dropdown-item>Make a copy</sui-dropdown-item>
          <sui-dropdown-item><sui-icon name="folder" />Move to folder</sui-dropdown-item>
          <sui-dropdown-item><sui-icon name="trash" />Move to trash</sui-dropdown-item>
          <sui-dropdown-divider/>
          <sui-dropdown-item>Download as...</sui-dropdown-item>
        </sui-dropdown-menu>
      </sui-dropdown-item>

      <sui-dropdown-item>Save as... <span class="description">ctrl + s</span></sui-dropdown-item>
      <sui-dropdown-item>Rename <span class="description">ctrl + r</span></sui-dropdown-item>
    </sui-dropdown-menu>
  </sui-dropdown>
</template>
```

image

I didn't test it with options prop
```
 <sui-dropdown
    :options="options" />

```

Also, this breaks Dropdown`s prop `closeOnBlur` behaviour because DropdownMenu has to handle `click` events on body.

### CODE REVIEW

**Comment by plugn** [comment]
image

**Comment by plugn** [comment]
oops. I need some help to fix tests

File: build/dev-server.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-require('./check-versions')()
+// require('./check-versions')()
```
Comment (mariolamacchia) [inline build/dev-server.js]: Why did you comment this out?

File: package.json [hunk 2]
```diff
@@ -105,7 +105,7 @@
     "sinon": "^2.1.0",
     "sinon-chai": "^2.8.0",
     "uglifyjs-webpack-plugin": "^1.1.4",
-    "url-loader": "^0.5.8",
+    "url-loader": "^1.1.2",
```
Comment (mariolamacchia) [inline package.json]: I don't think we need to update any dependency, do we?

File: docs/examples/modules/DropdownExample/index.js [hunk 3]
```diff
@@ -19,6 +20,11 @@ export default [
   {
     title: 'Types',
     examples: [
+      {
+        title: 'Dropdown Nested',
+        description: 'A dropdown with nested menu.',
+        component: DropdownNested,
+      },
```
Comment (mariolamacchia) [inline docs/examples/modules/DropdownExample/index.js]: Nit: I'd move this one at the end of the array, to keep the docs as similar as possible to the original website

File: src/lib/eventBus.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import Vue from 'vue';
+
+export const eventBus = new Vue();
```
Comment (mariolamacchia) [inline src/lib/eventBus.js]: Not sure I understand this, could you elaborate on what you are doing?

File: src/modules/Dropdown/DropdownMenu.jsx [hunk 5]
```diff
@@ -74,3 +90,5 @@ export default {
     parent: 'SuiDropdown',
   },
 };
+
+//         class={this.classes('menu', this.cssOpen, 'transition', this.animation)}
```
Comment (mariolamacchia) [inline src/modules/Dropdown/DropdownMenu.jsx]: Remove comment

File: src/lib/mixins/popupMixin.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        },
+      };
+      this.menu.$el.classList.remove('loading');
+      this.menuDirection = {
+        above: c.menu.offset.top - c.menu.height - this.$el.clientHeight >= 0,
+        below: c.menu.offset.top + c.menu.height < c.context.height,
+      };
+    },
+  },
+};
```
Comment (mariolamacchia) [inline src/lib/mixins/popupMixin.js]: In Semantic UI, a Popup is a different component. 

Also, I totally agree on splitting big files into smaller ones. However, this feature is already kind of big, so I wouldn't do it unless strictly necessary: it is harder to understand what you are doing and to identify what is making the tests fail.

File: src/lib/eventBus.js (hunk 4 above)
Comment (plugn) [reply src/lib/eventBus.js]: Here's an old-school approach of pub-sub in Vue. Now I should use this.$root.$emit() and this.$root.$on() instead

File: build/dev-server.js (hunk 1 above)
Comment (plugn) [reply build/dev-server.js]: comman yarn start was hanging at this point. This comment fixed it. Don't understand why :(

File: package.json (hunk 2 above)
Comment (plugn) [reply package.json]: Agree

File: src/lib/mixins/popupMixin.js (hunk 6 above)
Comment (plugn) [reply src/lib/mixins/popupMixin.js]: renamed to `dropdownMixin`

**Pull request closed without merging**